src/NServiceBus.RavenDB/Shutdown/ShutdownHook.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.RavenDB.Shutdown
+{
+    /// <summary>
+    /// This class is to be ported to the Endpoint level in NServiceBus core.
+    /// This allows us to fix the timeout shutdown bug without also needing to patch core.
+    /// </summary>
+    class ShutdownHook : IWantToRunWhenBusStartsAndStops
```

Comment:
scratch this, I'm going backward like a shrimp and is already behaving like I propose :smile: